app/models/compute_resources/foreman/model/ec2.rb

```diff
@@ -50,11 +50,41 @@ def find_vm_by_uuid(uuid)
       raise(ActiveRecord::RecordNotFound)
     end
 
-    def create_vm(args = { })
-      args = vm_instance_defaults.merge(args.to_h.symbolize_keys).deep_symbolize_keys
+
+    # Tag Example - <Fog::AWS::Compute::Tag key=nil, value=nil, resource_id=nil, resource_type=nil #>
+    def new_tag(attrs = {})
+      client.tags.new(attrs)
+    end
+
+    def parse_tags(args)
+      # Merge AWS EC2 tags
+      tags = {}
       if (name = args[:name])
-        args[:tags] = {:Name => name}
+        tags = {:Name => name}
+      end
+
+      [:tags].each do |collection|
+        nested_attrs = args.delete("#{collection}_attributes".to_sym)
+        args[collection] = nested_attributes_for(collection, nested_attrs.deep_symbolize_keys) if nested_attrs
       end
```

Comment:
Why is this a loop? Isn't this the same?
```suggestion
      nested_attrs = args.delete(:tags_attributes)
      args[:tags] = nested_attributes_for(:tags, nested_attrs.deep_symbolize_keys) if nested_attrs
```